src/workers/search/tests/get-matches.spec.js

```diff
@@ -71,5 +71,17 @@ describe("search", () => {
       });
       expect(matchLocations).toHaveLength(0);
     });
+
+    // regression test for #6896
+    it("doesn't crash on the regex 'a*'", () => {
```

Comment:
If you comment out the fix and run this test it will crash with an out of memory error.  I couldn't figure out a way to prevent the crash and fail more elegantly.